packages/logging/src/v2/config_service_v2_client.js

```diff
@@ -100,27 +100,94 @@ function ConfigServiceV2Client(gaxGrpc, grpcClients, opts) {
   var self = this;
 
   this.auth = gaxGrpc.auth;
-  var configServiceV2Stub = gaxGrpc.createStub(
-      grpcClients.google.logging.v2.ConfigServiceV2,
-      opts);
+
   var configServiceV2StubMethods = [
     'listSinks',
     'getSink',
     'createSink',
     'updateSink',
     'deleteSink'
   ];
+
+  var configServiceV2Stub;
+
   configServiceV2StubMethods.forEach(function(methodName) {
-    self['_' + methodName] = gax.createApiCall(
-      configServiceV2Stub.then(function(configServiceV2Stub) {
-        return function() {
-          var args = Array.prototype.slice.call(arguments, 0);
-          return configServiceV2Stub[methodName].apply(configServiceV2Stub, args);
-        };
-      }),
-      defaults[methodName],
-      PAGE_DESCRIPTORS[methodName]);
+    self['_' + methodName] = function() {
+      var args = [].slice.call(arguments);
+      var cb;
+
+      if (typeof args[args.length - 1] === 'function') {
+        cb = args.pop();
+      }
+
+      var promise = getStub().then(function(stub) {
+        var apiCall = gax.createApiCall(
+          configServiceV2Stub.then(function(configServiceV2Stub) {
+            return function() {
+              var args = Array.prototype.slice.call(arguments, 0);
+              var reqOpts = args[0];
+
+              try {
+                reqOpts = replaceProjectIdToken(reqOpts, self.auth.projectId);
+              } catch(e) {
+                var callback = args[args.length - 1];
+                callback(e);
+                return;
+              }
+
+              return configServiceV2Stub[methodName].apply(configServiceV2Stub, args);
```

Comment:
@stephenplusplus I think this is where that response array is coming from? Since promises can only resolve with a single parameter, I'm guessing when we run the API method in promise mode, it boxes everything in